pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
Sorry, sent the previous unfinished comment accidentally, fat fingers...

Should we take the mvn example input and execute

/bin/sh -c 'mvn "-Dapp=$(POD_NAME)"'

?
On Feb 1, 2016 7:49 PM, "Rodolfo Carvalho" <rcarvalh@redhat.com> wrote:

> Could you explain a bit more what is the "proper shell escaping"?
>
> In the `mvn` example there are no explicit quotes.
>
> I can imagine the problem to avoid is $POD_NAME expanding to a string that
> contains spaces.
> On Feb 1, 2016 7:00 PM, "Clayton Coleman" <notifications@github.com>
> wrote:
> >
> > In pkg/build/api/v1beta3/types.go:
> >
> > > @@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
> > >  ForcePull bool `json:"forcePull,omitempty" description:"forces the
> source build to pull the image if true"`
> > >  }
> > >
> > > +// BuildPostCommitSpec holds a build post commit hook specification.
> > > +type BuildPostCommitSpec struct {
> > > + // Shell is an action that executes a shell script in a container
> > > + // running the build output image.
> > > + Shell string `json:"shell,omitempty" description:"shell script to be
> executed in a container running the build output image"`
> >
> > I agree concatenation is not valuable in and of itself - but I'd like to
> hear counterarguments to the arguments handling issue if you have them. I
> think shellArgs: ["bundle", "exec, "rake", "test"] is at not quite as
> obvious as shell: "bundle exec rake test", but shellArgs: ["mvn",
> "-Dapp=$(POD_NAME)"] is clearer than shell: "mvn -Dapp=$(POD_NAME)" which
> could easily have spaces and be wrong.
> >
> > —
> > Reply to this email directly or view it on GitHub.
>